File: go/nth-prime/nth_prime.go
package prime

func Nth(n int) (int, bool) {
	if n < 1 {
        return 0, false
    }
	primes := []int{2}
    numberToCheck := 3
	for len(primes) < n {
        isPrime := true
        for _, prime := range primes {
            if numberToCheck % prime == 0 {
                isPrime = false
                numberToCheck++
                break
            }
        }
    	if isPrime {
            primes = append(primes, numberToCheck)
        	numberToCheck++
        }
    }
	return primes[n - 1], true
}

File: cpp/nth-prime/nth_prime.cpp
#include "nth_prime.h"
#include <vector>
#include <algorithm>

namespace nth_prime {
    int nth(int n) {
        if (n < 1) {
            throw std::domain_error("input must be positive");
        }
        std::vector<int> primes {2};
        int number{3};
        while (int(primes.size()) < n) {
            if (std::all_of(primes.begin(), primes.end(), [&number](int i){return number % i != 0;})) {
                primes.emplace_back(number);
            }
            number++;
        }
        return primes.back();
    }
}  // namespace nth_prime
